src/internal/m365/collection/drive/collections_tree_test.go

```diff
@@ -1128,14 +1170,22 @@ func (suite *CollectionsTreeUnitSuite) TestCollections_EnumeratePageOfItems_fold
 			err := c.enumeratePageOfItems(
 				ctx,
 				test.tree,
-				test.limiter,
-				&driveEnumerationStats{},
 				drv,
 				test.page,
+				test.limiter,
 				counter,
 				fault.New(true))
+
 			test.expect.err(t, err, clues.ToCore(err))
-			assert.Equal(t, test.expect.treeSize, test.tree.countFolders(), "count folders in tree")
+			if test.expect.shouldHitLimit {
+				assert.ErrorIs(t, err, errHitLimit, clues.ToCore(err))
+			}
+
+			assert.Equal(
+				t,
+				test.expect.treeSize,
+				len(test.tree.tombstones)+len(test.tree.folderIDToNode),
```

Comment:
nit: please use well-defined functions where possible instead of internal struct fields
```suggestion
				len(test.tree.tombstones)+test.tree.countLiveFolders,
```